vulnerabilities/improvers/default.py

```diff
@@ -0,0 +1,69 @@
+from typing import Iterable
+from typing import List
+from itertools import chain
+
+from django.db.models.query import QuerySet
+from packageurl import PackageURL
+
+from vulnerabilities.data_inference import Improver
+from vulnerabilities.data_inference import Inference
+from vulnerabilities.data_inference import MAX_CONFIDENCE
+from vulnerabilities.data_source import AdvisoryData
+from vulnerabilities.data_source import AffectedPackage
+from vulnerabilities.models import Advisory
+
+
+class DefaultImprover(Improver):
+    """
+    Generate a translation of Advisory data - returned by the importers - into
+    full confidence inferences. These are basic database relationships for
+    unstructured data present in the Advisory model without any other
+    information source.
+    """
+
+    @property
+    def interesting_advisories(self) -> QuerySet:
+        return Advisory.objects.all()
+
+    def get_inferences(self, advisory_data: AdvisoryData) -> Iterable[Inference]:
+        for affected_package in advisory_data.affected_packages:
+            affected_purls, fixed_purl = get_exact_purls(affected_package)
+            yield Inference(
+                aliases=advisory_data.aliases,
+                confidence=MAX_CONFIDENCE,
+                summary=advisory_data.summary,
+                affected_purls=affected_purls,
+                fixed_purl=fixed_purl,
+                references=advisory_data.references,
+            )
+
+
+def get_exact_purls(affected_package: AffectedPackage) -> (List[PackageURL], PackageURL):
+    """
+    Return purls for fixed and affected packages contained in the given
+    AffectedPackage disregarding any ranges.
+
+    Only exact version constraints (ie with an equality) are considered
+    For eg:
+    >>> purl = {"type": "turtle", "name": "green"}
+    >>> vers = "vers:npm/<1.0.0 | >=2.0.0 | <3.0.0"
+    >>> affected_package = AffectedPackage.from_dict({
+    ...     "package": purl,
+    ...     "affected_version_range": vers,
+    ...     "fixed_version": "5.0.0"
+    ... })
+    >>> get_exact_purls(affected_package)
+    ([PackageURL(type='turtle', namespace=None, name='green', version='2.0.0', qualifiers={}, subpath=None)], PackageURL(type='turtle', namespace=None, name='green', version='5.0.0', qualifiers={}, subpath=None))
+    """
+    affected_purls = set()
```

Comment:
IMHO we could refactor this as version resolution procedure using this kind of logic:
```
>>> from univers import *
>>> from univers.version_range import *
>>> VersionRange.from_string("vers:npm/<1.0.0 | >=2.0.0 | <3.0.0")
NpmVersionRange(constraints=(VersionConstraint(comparator='<', version=SemverVersion(string='1.0.0')), VersionConstraint(comparator='>=', version=SemverVersion(string='2.0.0')), VersionConstraint(comparator='<', version=SemverVersion(string='3.0.0'))))
>>> vr= VersionRange.from_string("vers:npm/<1.0.0 | >=2.0.0 | <3.0.0")
>>> range_versions = [c.version for c in vr.constraints]
>>> range_versions
[SemverVersion(string='1.0.0'), SemverVersion(string='2.0.0'), SemverVersion(string='3.0.0')]
>>> resolved_versions = [v for v in range_versions if v in vr]
>>> resolved_versions
[SemverVersion(string='2.0.0')]
>>> 
```